BeeTracker/app/src/main/java/com/blueprint/foe/beetracker/BeeAlertDialog.java

```diff
@@ -0,0 +1,73 @@
+package com.blueprint.foe.beetracker;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.app.DialogFragment;
+import android.app.Fragment;
+import android.content.Context;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+/**
+ * A generic dialog that should be modified for each type of dialog using the Bundle passed into
+ * onCreateDialog.
+ */
+public class BeeAlertDialog extends DialogFragment {
+    public static final String IMAGE_SRC = "IMAGE_SRC";
+    public static final String HEADING = "HEADING";
+    public static final String PARAGRAPH = "PARAGRAPH";
+    BeeAlertDialogListener mListener;
+
+    public interface BeeAlertDialogListener {
```

Comment:
I usually separate listeners and put them in their own `listeners` package.

Reply:
So you're not getting super verbose like `BeeAlertDialog.BeeAlertDialogListener`